prov/efa/test/efa_unit_test_device.c

```diff
@@ -0,0 +1,35 @@
+#include "efa_unit_tests.h"
+
+int efa_device_construct(struct efa_device *efa_device,
+			 int device_idx,
+			 struct ibv_device *ibv_device);
+
+/*
+ * test the error handling path of efa_device_construct()
+ */
+void test_efa_device_construct_error_handling()
+{
+	int ibv_err = 4242;
+	struct ibv_device **ibv_device_list;
+	struct efa_device efa_device = {0};
+
+	ibv_device_list = ibv_get_device_list(&g_device_cnt);
+	if (ibv_device_list == NULL) {
+		skip();
```

Comment:
Does this test depend on EFA device? If so, I think we should assert ibv_device_list is not NULL here.

Reply:
> It was trying to skip the test on a system without efa device. 

Maybe we should do this for all unit tests since currently they all require EFA device. It's odd to make this test special. Skipping it might hide real problems, i.e. EFA device is not registered properly and thus not discoverable by `ibv_get_device_list`.